src/main/java/cd/connect/openapi/DartV3ApiGenerator.java

```diff
@@ -213,6 +213,7 @@ private void correctInternals(CodegenModel model, CodegenProperty cp) {
           if (cm.vars != null) {
             cm.vars.forEach(cp -> {
               CodegenProperty correctingSettings = cp;
+              cp.vendorExtensions.put("generateNullValuesToJson", Boolean.parseBoolean((String) additionalProperties.get("generateNullValuesToJson")));
```

Comment:
if the additional properties have that value, it should be available as just hash-generateNullValuesToJson anywhere - all additional properties are normally available globally. Is the intention to allow for a vendor extension per model and a global one if set as an additional property? Thats what I am guessing. If so, we should probably write it up and put an x- in front of it (even though this strictly isn't required any longer).

Reply:
My intention was about that it have to be global accessible. We would like to pass as a generator argument as additional property, so it is not from yaml file. Maybe there is another place for it but I've found only that one to set. 



```
//////////////////////////////////////////////////////////////////////////////
// G E N E R A T E D    C L A S S
// ------------------------------
// Factory expression: {#application}
// Path expression: 'lib/'+#path(#application.actor.name)+'/rest/'+#path(#application.actor.name)+'.dart'
// Overwrite: true
// Template name: flutter/lib/rest/openapi_generator_config.dart.hbs
// Application: SkillMatrix::actor::hrEmployee::HREmployeeActor

import 'package:openapi_generator_annotations/openapi_generator_annotations.dart';

@Openapi(
    additionalProperties: AdditionalProperties(
          pubName: 'hREmployeeActor', 
          pubAuthor: 'Judo Generator', 
          generateNullValuesToJson: true
    ),
    inputSpecFile: 'lib/hremployee_actor/rest/hremployee_actor.yaml',
    alwaysRun: true,
    overwriteExistingFiles: true,
    generatorName: Generator.DART2_API,
    outputDirectory: 'lib/hremployee_actor/rest',
    reservedWordsMappings: {
        '__identifier' : 'internal__identifier',
        '__signedIdentifier' : 'internal__signedIdentifier',
        '__referenceId' : 'internal__referenceId',
        '__updateable' : 'internal__updatable',
        '__deleteable' : 'internal__deletable'
    },
    fetchDependencies: false,
    runSourceGenOnOutput: false
)
class HREmployeeActor extends OpenapiGeneratorConfig {}
```

Reply:
Can you give me an example please? Maybe my understanding is wrong about it

Reply:
but if you have done exactly that - pass a generateNullValuesToJson in your additional properties then it becomes a global to the mustache templates. You don't then need to put it in each parameter. For example, change the dataType line from your other PR to

`{{{dataType}}}{{#nullSafe}}?{{/nullSafe}} {{{name}}}{{#defaultValue}} = {{{defaultValue}}}{{/defaultValue}}; {{sausage}}
`

and then add an additional property sausage: 'pavlova' to your code above (what is this code by the way?) - and you will see it turn up everywhere. The mustache template generator just walks up the tree trying to find a match.

Reply:
Further you shouldn't put "true" or "false" on a boolean as it checks for existence not truthy. If false you need to not even put the entry in.